lib/Parser/Parser.yy

```diff
@@ -50,155 +50,608 @@
+    | parameter_list comma
+    ;
+
+parameter:
+      ident colon type
+    | ident colon type equal expression
+    ;
+
+function_body:
+      block
+    | semi
+    ;
+
+block:
+      lBrace statement_list rBrace
+    ;
+
+statement_list:
+      /* empty */
+    | statement_list statement
+    ;
+
+statement:
+      block
+    | expression_stmt semi
+    | var_stmt
+    | let_stmt
+    | return_stmt
     | if_stmt
     | while_stmt
-    | return_stmt
+    | for_stmt
+    | break_stmt
+    | continue_stmt
+    | assignment_stmt
+    | semi
+    ;
+
+expression_stmt:
+      conditional_expression_stmt
+    ;
+
+conditional_expression_stmt:
+  logical_or_expression_stmt
+    | logical_or_expression_stmt question expression colon conditional_expression_stmt %prec TERNARY
+      { std::cerr << "Parsed ternary expression" << std::endl; }
+    ;
+
+logical_or_expression_stmt:
+  logical_or_expression_stmt orOp logical_and_expression_stmt
+      { std::cerr << "Parsed logical or" << std::endl; }
+    | logical_and_expression_stmt
+
+logical_and_expression_stmt:
+  logical_and_expression_stmt andOp conditional_expression_stmt
+      { std::cerr << "Parsed logical and" << std::endl; }
+    | function_call_stmt
+    ;
+
+function_call_stmt:
+  namespaced_identifier
+    | lParen expression_stmt rParen
+    | function_call_stmt function_call
+    ;
+
+function_call:
+  function_template_arguments lParen argument_list_opt rParen %prec POSTFIX
+    ; 
+
+function_template_arguments:
+  /* empty */
+      { std::cerr << "Parsed empty function template arguments" << std::endl; }
+    | coloncolonLt type_list gtOp
+      { std::cerr << "Parsed function template arguments" << std::endl; }
+    ;
+
+var_stmt:
+      varKw ident type_opt initializer_opt semi
+    { std::cerr << "Parsed var statement" << std::endl; }
+    ;
+
+type_opt:
+      /* empty */
+    | colon type
+    ;
+
+initializer_opt:
+      /* empty */
+    | equal expression
+    ;
+
+let_stmt:
+      letKw ident type_opt equal expression semi
+    { std::cerr << "Parsed let statement" << std::endl; }
+    ;
+
+return_stmt:
+      returnKw expression_opt semi
+    { std::cerr << "Parsed return statement" << std::endl; }
+    ;
+
+expression_opt:
+      /* empty */
+    | expression
+    ;
+
+if_stmt:
+      ifKw expression block else_opt
+    { std::cerr << "Parsed if statement" << std::endl; }
+    ;
+
+else_opt:
+      /* empty */
+    | elseKw statement
+    ;
+
+while_stmt:
+      whileKw expression block
+    { std::cerr << "Parsed while statement" << std::endl; }
+    ;
+
+for_stmt:
+      forKw ident inKw expression block
+    { std::cerr << "Parsed for statement" << std::endl; }
+    ;
+
+break_stmt:
+      breakKw semi
+    { std::cerr << "Parsed break statement" << std::endl; }
+    ;
+
+continue_stmt:
+      continueKw semi
+    { std::cerr << "Parsed continue statement" << std::endl; }
+    ;
+
+assignment_stmt:
+      expression equal expression semi
```

Comment:
this will clearly also have an ambiguity sadly. we should probably have an `lvalue_expression` or `untyped_expression` (cannot be a struct ctor if we can't possibly know the type in this context)